DataWindowQuery : Add node for querying DataWindow

ericmehl opened the pull request:
This does what it says in the title, along with a couple of additional output plugs for the data window's center and size. This fixes #5637.

### Checklist ###

- [X] I have read the [contribution guidelines](https://github.com/GafferHQ/gaffer/blob/main/CONTRIBUTING.md).
- [X] I have updated the documentation, if applicable.
- [X] I have tested my change(s) in the test suite, and added new test cases where necessary.
- [X] My code follows the Gaffer project's prevailing coding style and conventions.

murraystevenson commented on python/GafferImageUI/DataWindowQueryUI.py:
```diff
@@ -0,0 +1,95 @@
+
+Gaffer.Metadata.registerNode(
+
+	GafferImage.DataWindowQuery,
```
We're missing a node description here.

murraystevenson commented on python/GafferImageUI/DataWindowQueryUI.py:
```diff
@@ -0,0 +1,95 @@
+			"""
+			The view to query.
+			""",
+
```
This plug might benefit from some additional metadata:

```
"nodule:type" : "",
"plugValueWidget:type" : "GafferImageUI.ViewPlugValueWidget",
"viewPlugValueWidget:allowUseCurrentContext" : True,
```

murraystevenson commented on python/GafferImageUI/DataWindowQueryUI.py:
```diff
@@ -0,0 +1,95 @@
+			The data window of the image.
+			""",
+			"nodule:type" : "GafferUI::CompoundNodule",
+			"noduleLayout:spacing" : 0.4,
```
I don't think this `spacing` metadata is required? We don't usually put space between the nodules of related child plugs, such as `min` and `max` here. See `FormatQuery.format` for prior art.

ericmehl commented:
Thanks for looking over that @murraystevenson! I fixed up the missing UI elements and in https://github.com/GafferHQ/gaffer/pull/6676/commits/633b9725a2fcd6125a0b375efae808151ffaf5b9 I added an `Output` section similar to FormatQuery.

ericmehl commented:
I pushed https://github.com/GafferHQ/gaffer/pull/6676/commits/4d1a867980e7d10b3257eb1371cc15eb49ea0e20 which changes the name of the section for output plugs to `Outputs` (from `Out`) to be in line with most of the other query nodes.

One thing I noticed when adding a DataWindowQuery that isn't yet connected to anything, the default data window shows up as this since it's an invalid `Box2i`. I don't know what use an unconnected query node would have, but is there any value to checking for invalid `Box2i` and setting the output plug to something different, like an empty `Box2i`?
<img width="1136" height="506" alt="image" src="https://github.com/user-attachments/assets/ede5dc8f-4d37-4853-885b-0d14c474994b" />

murraystevenson commented on Changes.md:
```diff
@@ -1,6 +1,11 @@
+Features
+--------
+
+- DataWindowQuery : Added a new node for querying image data windows.
```
As this was requested in #5637, we should be including that here and in the commit message.

murraystevenson reviewed (changes requested):
Thanks Eric! I've noted a couple of minor things inline, but otherwise we're looking pretty good.

> One thing I noticed when adding a DataWindowQuery that isn't yet connected to anything, the default data window shows up as this since it's an invalid Box2i. I don't know what use an unconnected query node would have, but is there any value to checking for invalid Box2i and setting the output plug to something different, like an empty Box2i?

We do something along those lines in [BoundQuery](https://github.com/GafferHQ/gaffer/blob/aeffc81abdb3bfc97ba43c74fc959691487b2a79/src/GafferScene/BoundQuery.cpp#L350) to attempt to avoid showing the min/max of a default Box3f, but I don't think it's necessary here, and may be counter-productive if we replace all empty data windows with something like a `Box2i( V2i( 0 ) )`. 

What might be useful is an `empty` BoolPlug that outputs whether the queried data window is empty, but I don't think we need that today as you can currently query that from the Box2i output from the `dataWindow` plug.

ericmehl commented on include/GafferImage/DataWindowQuery.h:
```diff
@@ -0,0 +1,88 @@
+		const Gaffer::Box2iPlug *dataWindowPlug() const;
+
+		Gaffer::V2iPlug *centerPlug();
+		const Gaffer::V2iPlug *centerPlug() const;
```
Oh right, that should definitely be float. Fixed in https://github.com/GafferHQ/gaffer/pull/6676/commits/5b45f70e7b4d9579915ae0b825b84a2d9cac8386
